weave/trace_server_bindings/remote_http_trace_server.py

```diff
@@ -129,19 +134,51 @@ def set_auth(self, auth: tuple[str, str]) -> None:
         retry_error_callback=_log_failure,
         reraise=True,
     )
+    def _send_batch_to_server(self, encoded_data: bytes) -> None:
+        """Send a batch of data to the server with retry logic.
+
+        This method is separated from _flush_calls to avoid recursive retries.
+        """
+        r = requests.post(
+            self.trace_server_url + "/call/upsert_batch",
+            data=encoded_data,  # type: ignore
+            auth=self._auth,
+            timeout=DEFAULT_TIMEOUT,
+        )
+        # TODO: I think this handles the case where len(batch) == 1 and
+        # encoded_bytes > self.remote_request_bytes_limit case, but it feels weird to
+        # have to send it to the server to find out.  We should just save the network
+        # cost if we know we can't send it...
+        if r.status_code == 413:
+            # handle 413 explicitly to provide actionable error message
+            reason = json.loads(r.text)["reason"]
+            raise requests.HTTPError(f"413 Client Error: {reason}", response=r)
+        r.raise_for_status()
+
     def _flush_calls(
         self,
         batch: list,
         *,
         _should_update_batch_size: bool = True,
     ) -> None:
+        """Process a batch of calls, splitting if necessary and sending to the server.
+
+        This method handles the logic of splitting batches that are too large,
+        but delegates the actual server communication (with retries) to _send_batch_to_server.
+        """
         if len(batch) == 0:
             return
 
         data = Batch(batch=batch).model_dump_json()
         encoded_data = data.encode("utf-8")
         encoded_bytes = len(encoded_data)
 
+        if len(batch) == 1 and encoded_bytes > self.remote_request_bytes_limit:
+            raise ValueError(
+                f"Single call size ({encoded_bytes}) is too large to send.  Max size: "
+                f"{self.remote_request_bytes_limit} bytes.  Unable to send."
+            )
+
```

Comment:
This saves on an unnecessary network request because we know the server will reject it anyways

Reply:
I think `remote_request_bytes_limit` is a best-guess and not a hard rule (and has changed with server implementations). I think it would be better to at least attempt the request

Reply:
ok, will remove this check